### PR TITLE
Fix strict variables

### PR DESCRIPTION
**Pull request by coreone**
* Fix strict variables for defaults in params.pp
* Fix versioncmp for Puppet 4 in params.pp
* Test for undefined variable in snmp.conf.erb
* Add missing trap_service_name for Debian in params.pp

### CODE REVIEW

**Comment by coreone** [comment]
Fixes Travis errors from #97

**Comment by coreone** [comment]
@gusnuf @razorsedge refactored to comply with CONTRIBUTING.  I will re-submit another PR for the master/agentx changes once this PR is merged.

**Comment by coreone** [comment]
If support for top-scope variables was no longer needed, something like #85 would make much more sense since it makes `params.pp` much cleaner.  If that were the case, I would rebase this and just fix the RedHat-specific variables and the `versioncmp` issue in `params.pp`.

File: manifests/params.pp [hunk 1]
```diff
@@ -358,6 +438,7 @@
       $client_package_name      = 'snmp'
       $client_config            = '/etc/snmp/snmp.conf'
 
+      $trap_service_name        = 'snmptrapd'
```
Comment (razorsedge) [inline manifests/params.pp]: What does this have to do with strict vars?  I believe it is missing because tested versions of Debian do not have it.

File: manifests/params.pp [hunk 2]
```diff
@@ -302,7 +382,7 @@
           $service_config_perms = '0600'
         }
         default: {
-          if $majdistrelease <= '5' {
+          if versioncmp($majdistrelease, '5') <= 0 {
```
Comment (coreone) [inline manifests/params.pp]: versioncmp is part of Puppet since at least 3.5:

https://docs.puppet.com/puppet/3.5/function.html#versioncmp

File: manifests/params.pp (hunk 1 above)
Comment (coreone) [inline manifests/params.pp]: In `init.pp`, `$trap_service_name` is defaulted to `$snmp::params::trap_service_name`, so if that is happening on Debian, the default will not exist in params, which can lead to warnings in P4 about undefined variables.

File: manifests/params.pp (hunk 2 above)
Comment (coreone) [inline manifests/params.pp]: Additionally, `getvar` is in stdlib 2.3.0, so `metadata.json` should not need editing for that either:

https://github.com/puppetlabs/puppetlabs-stdlib/tree/2.3.0/lib/puppet/parser/functions

**Comment by coreone** [comment]
@razorsedge I believe I answered everything inline...

File: manifests/params.pp (hunk 1 above)
Comment (razorsedge) [inline manifests/params.pp]: In that case, lets set it to empty `''`.

File: manifests/params.pp (hunk 2 above)
Comment (razorsedge) [inline manifests/params.pp]: Excellent.

**Comment by coreone** [comment]
@razorsedge should be all set.  Set it to empty string instead of `snmptrapd`

**Comment by coreone** [comment]
empty string 👎 , undef 👍
